modules/apps/content-dashboard/content-dashboard-document-library-test/src/testIntegration/java/com/liferay/content/dashboard/document/library/internal/item/action/provider/test/ViewFileVersionContentDashboardItemVersionActionProviderTest.java

```diff
@@ -0,0 +1,176 @@
+import com.liferay.portal.kernel.util.ContentTypes;
+import com.liferay.portal.kernel.util.HtmlUtil;
+import com.liferay.portal.kernel.util.JavaConstants;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.MimeTypesUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/**
+ * @author Jürgen Kappler
+ */
+@RunWith(Arquillian.class)
+public class ViewFileVersionContentDashboardItemVersionActionProviderTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup(
+			TestPropsValues.getCompanyId(), TestPropsValues.getUserId(), 0);
+	}
+
+	@Test
+	public void testGetContentDashboardItemAction() throws Exception {
+		_fileEntry = _dlAppLocalService.addFileEntry(
+			RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+			_group.getGroupId(), DLFolderConstants.DEFAULT_PARENT_FOLDER_ID,
+			RandomTestUtil.randomString() + "." + ContentTypes.IMAGE_JPEG,
+			MimeTypesUtil.getExtensionContentType(ContentTypes.IMAGE_JPEG),
+			new byte[0], null, null,
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest();
+
+		MockLiferayResourceRequest mockLiferayResourceRequest =
+			new MockLiferayResourceRequest();
+
+		mockLiferayResourceRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, _getThemeDisplay(mockHttpServletRequest));
+
+		mockHttpServletRequest.setAttribute(
+			JavaConstants.JAVAX_PORTLET_REQUEST, mockLiferayResourceRequest);
+
+		mockHttpServletRequest.setAttribute(
+			JavaConstants.JAVAX_PORTLET_RESPONSE,
+			new MockLiferayPortletRenderResponse());
+
+		ContentDashboardItemVersionAction contentDashboardItemVersionAction =
+			_contentDashboardItemVersionActionProvider.
+				getContentDashboardItemVersionAction(
+					_fileEntry.getFileVersion(), mockHttpServletRequest);
+
+		Assert.assertEquals(
+			"view", contentDashboardItemVersionAction.getName());
+		Assert.assertEquals(
+			"view", contentDashboardItemVersionAction.getIcon());
+		Assert.assertEquals(
+			_language.get(LocaleUtil.US, "view"),
+			contentDashboardItemVersionAction.getLabel(LocaleUtil.US));
+
+		String url = contentDashboardItemVersionAction.getURL();
+
+		Assert.assertTrue(Validator.isNotNull(url));
```

Comment:
Maybe we can use Assert.assetNotNull?

Reply:
My bad @cgoncas, will also change this

Reply:
@cgoncas, added a new commit to standardize all tests https://github.com/liferay-tango/liferay-portal/pull/2917/commits/7ab79ec3d7e954efbb508b4da094c190ef7d7f71. Most likely we were checking for empty strings, but the next lines should always catch this